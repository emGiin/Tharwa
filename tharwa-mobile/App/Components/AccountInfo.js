import React from 'react'
import { TouchableOpacity, View } from 'react-native'
import { Text } from 'native-base'
import I18n from 'react-native-i18n'
import { formatMoney } from '../Transforms'

// Styles
import styles from './Styles/AccountInfoStyles'

export const AccountInfo = ({ account, type, label, symbol, onPress }) => {
  const getText = () => {
    if (account) {
      const { status } = account
      if (!status) return `${formatMoney(account.amount)} ${symbol}`
      if (status === 'requested') return I18n.t('accountRequested')
      if (status === 'blocked') return I18n.t('accountBlocked')
    } else return I18n.t('accountRequest')
  }

  return (
    <View style={styles.page}>
      <TouchableOpacity
        onPress={onPress}
        activeOpacity={0.9}
        style={styles.contentContainer}>
        <View style={styles.amountContainer}>
          <Text style={styles.amount}>
            {getText()}
          </Text>
        </View>
        <Text style={styles.account}>{label}</Text>
      </TouchableOpacity>
    </View >
  )
}