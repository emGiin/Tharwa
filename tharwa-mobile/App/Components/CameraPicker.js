import React, { Component } from 'react'
import { TouchableOpacity } from 'react-native'
import ImagePicker from 'react-native-image-picker';

class CameraPicker extends Component {

  takePicture = () => {
    const options = { quality: 0.5, base64: true };
    this.camera.capture({ metadata: options })
      .then(data => {
        this.props.input.onChange(data);
        this.props.onCapture(data)
      }).catch(err => console.error(err));
  }

  onPress = () => {
    var options = {
      title: 'Select Avatar',
      customButtons: [
        { name: 'fb', title: 'Choose Photo from Facebook' },
      ],
      permissionDenied: {
        title: 'titlee',
        text: 'textttt',
        reTryTitle: '3awed',
        okTitle: 'pff'
      },
      maxWidth: 1024,
      maxHeight: 1024,
      cameraType: 'front',
      mediaType: 'photo',
      quality: 0.5
    };
    ImagePicker.launchCamera(options, ({ data }) => {
      if (data) {
        const base64 = `data:image/jpeg;base64,${data}`;
        this.props.input.onChange(base64);
        this.props.onCapture(base64)
      }
    });
  }

  render() {
    return (
      <TouchableOpacity style={this.props.style} onPress={this.onPress.bind(this)}>
        {this.props.buttonComponent()}
      </TouchableOpacity>
    )
  }
}

export { CameraPicker }
