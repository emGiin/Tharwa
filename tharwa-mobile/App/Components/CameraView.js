import React, { Component } from 'react'
import { View, TouchableOpacity } from 'react-native'
import { Icon } from 'native-base'
import Camera from 'react-native-camera';
import styles from './Styles/CameraViewStyle'

class CameraView extends Component {
  takePicture = () => {
    const options = { quality: 0.5, base64: true };
    this.camera.capture({ metadata: options })
      .then(data => { this.props.input.onChange(data); this.props.onCapture(data) })
      .catch(err => console.error(err));
  }

  render() {
    const { previousPage } = this.props;
    return (
      <View style={styles.container}>
        <TouchableOpacity style={styles.backIcon} onPress={previousPage}>
          <Icon
            name={'ios-arrow-dropleft'}
            style={{ color: '#fff', fontSize: 40 }} />
        </TouchableOpacity>
        <Camera
          ref={cam => this.camera = cam}
          style={styles.preview}
          onBarCodeRead={() => { }}
          aspect={Camera.constants.Aspect.fill}>
          <TouchableOpacity onPress={this.takePicture}>
            <Icon
              name={'ios-camera'}
              style={{ color: '#fff', fontSize: 70 }} />
          </TouchableOpacity>
        </Camera>
      </View>
    )
  }
}

export { CameraView }
