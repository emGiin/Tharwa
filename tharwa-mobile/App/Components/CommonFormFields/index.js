export { default as EmailField } from './EmailField'
export { default as PasswordField } from './PasswordField'
export { default as InputField } from './InputField'
export { default as PickerField } from './InputField'