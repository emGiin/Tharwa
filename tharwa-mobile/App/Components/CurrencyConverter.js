import React from 'react'
import { View, Text } from 'react-native'
import { Icon } from 'native-base'
import { InputField, PickerField } from './'
import styles from './Styles/CurrencyConverterStyles'

const CurrencyConverter = ({
  from, to, fromCurrencies, toCurrencies,
  handleInputChange, handleSelectChange
}) => (
    <View>
      <Text style={styles.title}>{"Convertisseur "}</Text>
      <View style={styles.valueContainer}>
        <InputField
          keyboardType={'numeric'} returnKeyType={'done'}
          input={{ onChangeText: handleInputChange('from'), value: from.value }} />
        <Icon name={'ios-swap'} style={styles.icon} />
        <InputField
          keyboardType={'numeric'} returnKeyType={'done'}
          input={{ onChangeText: handleInputChange('to'), value: to.value }} />
      </View>
      <View style={styles.currenciesContainer}>
        <PickerField
          options={fromCurrencies} initialValue={fromCurrencies[0]}
          input={{ onChange: handleSelectChange('from'), value: from.currency }} />
        <View style={styles.separator} />
        <PickerField
          options={toCurrencies} initialValue={toCurrencies[0]}
          input={{ onChange: handleSelectChange('to'), value: to.currency }} />
      </View>
    </View>
  )

export { CurrencyConverter }