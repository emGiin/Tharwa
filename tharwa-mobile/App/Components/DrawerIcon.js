import React, { Component } from 'react';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';

class DrawerIcon extends Component {

  render() {
    const { focused, tintColor, icon, unselectedIcon, size = 26 } = this.props
    return (
      <Icon
        name={focused ? icon : (unselectedIcon || `${icon}-outline`)}
        size={size}
        style={{ color: tintColor }}
      />
    );
  }
}

export { DrawerIcon }