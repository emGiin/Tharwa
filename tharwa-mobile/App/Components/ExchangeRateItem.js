import React from 'react'
import { View, Text, TouchableOpacity } from 'react-native'
import { Icon } from 'native-base'
import styles from './Styles/ExchangeRateItemStyles'

const ExchangeRateItem = ({ from, to, value, onPress }) => (
  <TouchableOpacity
    onPress={() => onPress(from, to, value)}
    activeOpacity={0.5}
    style={styles.container}>
    <View style={styles.subContainer}>
      <Text style={styles.currency}>{from}</Text>
      <Text style={styles.value}>1</Text>
    </View>
    <View style={[styles.subContainer, { width: 50 }]}>
      <Icon name={'ios-swap'} style={styles.swapIcon} />
    </View>
    <View style={styles.subContainer}>
      <Text style={styles.currency}>{to}</Text>
      <Text style={styles.value}>{value}</Text>
    </View>
  </TouchableOpacity>
)


export { ExchangeRateItem }