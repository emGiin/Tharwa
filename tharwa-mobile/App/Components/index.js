export * from './AlertMessage'
export * from './DrawerButton'
export * from './FullButton'
export * from './RoundedButton'
export * from './FormFields'
export * from './Header'
export * from './NextPrevious'
export * from './LoadingDialog'
export * from './CameraPicker'
export * from './DrawerIcon'