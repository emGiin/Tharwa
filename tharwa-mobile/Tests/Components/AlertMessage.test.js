import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import { AlertMessage } from '../../App/Components'

describe('AlertMessage component', () => {
  let container, content;
  beforeAll(() => {
    container = shallow(<AlertMessage title='howdy' />)
    content = container.dive()
  })

  it('should render correctly if show is true', () => {
    expect(content).toMatchSnapshot()
  })


  it('should not render if show is false', () => {
    container.setProps({ show: false })
    container.update()
    expect(content).toMatchSnapshot()
  })
})
