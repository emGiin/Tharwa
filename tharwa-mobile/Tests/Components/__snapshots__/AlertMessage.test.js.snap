// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertMessage component should not render if show is false 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <View
    style={
      Array [
        Object {
          "justifyContent": "center",
          "marginVertical": 25,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#CCCCCC",
            "fontFamily": "Avenir-Book",
            "fontSize": 17,
            "fontWeight": "bold",
            "marginHorizontal": 10,
            "marginTop": 10,
            "textAlign": "center",
          }
        }
      >
        HOWDY
      </Text>
    </View>
  </View>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#CCCCCC",
              "fontFamily": "Avenir-Book",
              "fontSize": 17,
              "fontWeight": "bold",
              "marginHorizontal": 10,
              "marginTop": 10,
              "textAlign": "center",
            }
          }
        >
          HOWDY
        </Text>
      </View>,
      "style": Array [
        Object {
          "justifyContent": "center",
          "marginVertical": 25,
        },
        undefined,
      ],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#CCCCCC",
              "fontFamily": "Avenir-Book",
              "fontSize": 17,
              "fontWeight": "bold",
              "marginHorizontal": 10,
              "marginTop": 10,
              "textAlign": "center",
            }
          }
        >
          HOWDY
        </Text>,
        "style": Object {
          "alignItems": "center",
          "alignSelf": "center",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": false,
          "children": "HOWDY",
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "#CCCCCC",
            "fontFamily": "Avenir-Book",
            "fontSize": 17,
            "fontWeight": "bold",
            "marginHorizontal": 10,
            "marginTop": 10,
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": "HOWDY",
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "View",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#CCCCCC",
                "fontFamily": "Avenir-Book",
                "fontSize": 17,
                "fontWeight": "bold",
                "marginHorizontal": 10,
                "marginTop": 10,
                "textAlign": "center",
              }
            }
          >
            HOWDY
          </Text>
        </View>,
        "style": Array [
          Object {
            "justifyContent": "center",
            "marginVertical": 25,
          },
          undefined,
        ],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#CCCCCC",
                "fontFamily": "Avenir-Book",
                "fontSize": 17,
                "fontWeight": "bold",
                "marginHorizontal": 10,
                "marginTop": 10,
                "textAlign": "center",
              }
            }
          >
            HOWDY
          </Text>,
          "style": Object {
            "alignItems": "center",
            "alignSelf": "center",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": false,
            "children": "HOWDY",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "#CCCCCC",
              "fontFamily": "Avenir-Book",
              "fontSize": 17,
              "fontWeight": "bold",
              "marginHorizontal": 10,
              "marginTop": 10,
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": "HOWDY",
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "View",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`AlertMessage component should render correctly if show is true 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <View
    style={
      Array [
        Object {
          "justifyContent": "center",
          "marginVertical": 25,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#CCCCCC",
            "fontFamily": "Avenir-Book",
            "fontSize": 17,
            "fontWeight": "bold",
            "marginHorizontal": 10,
            "marginTop": 10,
            "textAlign": "center",
          }
        }
      >
        HOWDY
      </Text>
    </View>
  </View>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#CCCCCC",
              "fontFamily": "Avenir-Book",
              "fontSize": 17,
              "fontWeight": "bold",
              "marginHorizontal": 10,
              "marginTop": 10,
              "textAlign": "center",
            }
          }
        >
          HOWDY
        </Text>
      </View>,
      "style": Array [
        Object {
          "justifyContent": "center",
          "marginVertical": 25,
        },
        undefined,
      ],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#CCCCCC",
              "fontFamily": "Avenir-Book",
              "fontSize": 17,
              "fontWeight": "bold",
              "marginHorizontal": 10,
              "marginTop": 10,
              "textAlign": "center",
            }
          }
        >
          HOWDY
        </Text>,
        "style": Object {
          "alignItems": "center",
          "alignSelf": "center",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": false,
          "children": "HOWDY",
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "#CCCCCC",
            "fontFamily": "Avenir-Book",
            "fontSize": 17,
            "fontWeight": "bold",
            "marginHorizontal": 10,
            "marginTop": 10,
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": "HOWDY",
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "View",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#CCCCCC",
                "fontFamily": "Avenir-Book",
                "fontSize": 17,
                "fontWeight": "bold",
                "marginHorizontal": 10,
                "marginTop": 10,
                "textAlign": "center",
              }
            }
          >
            HOWDY
          </Text>
        </View>,
        "style": Array [
          Object {
            "justifyContent": "center",
            "marginVertical": 25,
          },
          undefined,
        ],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#CCCCCC",
                "fontFamily": "Avenir-Book",
                "fontSize": 17,
                "fontWeight": "bold",
                "marginHorizontal": 10,
                "marginTop": 10,
                "textAlign": "center",
              }
            }
          >
            HOWDY
          </Text>,
          "style": Object {
            "alignItems": "center",
            "alignSelf": "center",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": false,
            "children": "HOWDY",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "#CCCCCC",
              "fontFamily": "Avenir-Book",
              "fontSize": 17,
              "fontWeight": "bold",
              "marginHorizontal": 10,
              "marginTop": 10,
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": "HOWDY",
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "View",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
